spec/prometheus/client/data_stores/direct_file_store_spec.rb

```diff
@@ -267,6 +292,57 @@
     end
   end
 
+  context "with a metric that takes MOST_RECENT instead of SUM" do
+    it "reports the most recently written value from different processes" do
+      metric_store1 = subject.for_metric(
+        :metric_name,
+        metric_type: :gauge,
+        metric_settings: { aggregation: :most_recent }
+      )
+      metric_store2 = subject.for_metric(
+        :metric_name,
+        metric_type: :gauge,
+        metric_settings: { aggregation: :most_recent }
+      )
+
+      allow(Process).to receive(:pid).and_return(12345)
```

Comment:
I find this block kinda dense to read. Wondering if either [RSpec multiple return](https://relishapp.com/rspec/rspec-mocks/docs/configuring-responses/returning-a-value#specify-different-return-values-for-multiple-calls) syntax or just breaking it up with newlines would help.

I think the fact that the lines are all identical length somehow makes it harder to read.

Reply:
So this is one of the challenges of coming back to code 4 months later... I don't remember why this is written this way...

I'm not really sure why we have lines 312 and 313, for example...
I'm also not entirely sure why we need to be going back and forth with the process IDs. i think it's so the "pid change detector" doesn't change the file you're reading... let me take a stab at rewriting this...

Reply:
@Sinjo I've just reordered and separated the lines, hopefully it'll make a bit more sense (check the last commit, the fixup one)

Reply:
LGTM